Sparrow.ps1

```diff
@@ -143,12 +151,29 @@ Function Get-UALData {
         Y {$AppIdInvestigation = "Yes"}
         N {$AppIdInvestigation = "No"}
     }
+    
     If ($AppIdInvestigation -eq "Yes"){
         $SusAppId = Read-Host "Enter the application's AppID to investigate"
+        $AzureAppIds=Get-AzureADApplication -All
     } Else{
         Write-Host "Skipping AppID investigation"
     }
 
+     if ($AppIdInvestigation -eq "No")
+    {
+    $AppIdQuestion_GetAll = Read-Host 'Would you like to investigate all applications? This may take some time... Y/N'
+    Switch ($AppIdQuestion_GetAll){
+        Y {$AppIdQuestion_GetAll = "Yes"}
+        N {$AppIdQuestion_GetAll = "No"}
+    }
+    If ($AppIdQuestion_GetAll -eq "Yes"){
+        write-Host "Gathering Azure Application IDs..."
+        $AzureAppIds=Get-AzureADApplication -All:$true
```

Comment:
Recommend using `Get-AzureADServicePrincipal -All $true | Where-Object {$_.ServicePrincipalType -eq "Application"}` instead

Reply:
The requested change has been added. Also added comments to the file creation logic.